src/validate.rs

```diff
@@ -240,10 +241,21 @@ impl<Octs: AsRef<[u8]>> Key<Octs> {
         Octs: FromBuilder,
         Octs::Builder: EmptyBuilder + Composer,
     {
-        // Ensure there is a single line in the input.
-        let (line, rest) = dnskey.split_once('\n').unwrap_or((dnskey, ""));
-        if !rest.trim().is_empty() {
-            return Err(ParseDnskeyTextError::Misformatted);
+        // Skip leading comment lines (BIND uses these to record key timing
+        // data)
+        let mut line = dnskey;
+        while let Some((this_line, rest)) = line.split_once('\n') {
+            if !this_line.starts_with(';') {
+                // Ensure there is a single data line in the input.
+                if !rest.trim().is_empty() {
+                    return Err(ParseDnskeyTextError::Misformatted);
+                } else {
+                    line = this_line;
+                    break;
+                }
+            } else {
+                line = rest;
+            }
```

Comment:
I've updated my PR to account for this better.  Let me know if my new code works.